### PR TITLE
Add the ability to override the user agent for the oauth2 flow

### PR DESCRIPTION
**Pull request by deeringc**
… well as a scoped server test to verify.

### CODE REVIEW

File: Release/include/cpprest/oauth2.h [hunk 1]
```diff
@@ -436,23 +438,34 @@ class oauth2_config
...
-
-	/// <summary>
-	/// Set the web proxy object that will be used by token_from_code and token_from_refresh
-	/// </summary>
-	/// <param name="proxy">A reference to the web proxy object.</param>
-	void set_proxy(const web_proxy& proxy)
-	{
-		m_proxy = proxy;
-	}
+    /// <summary>
```
Comment (deeringc) [inline Release/include/cpprest/oauth2.h]: Just changed tabs to spaces for this, the rest of the file is all spaces.

File: Release/include/cpprest/oauth2.h [hunk 2]
```diff
@@ -436,23 +438,34 @@ class oauth2_config
...
+    /// Set the web proxy object that will be used by token_from_code and token_from_refresh
+    /// </summary>
+    /// <param name="proxy">A reference to the web proxy object.</param>
+    void set_proxy(const web_proxy& proxy)
+    {
+        m_proxy = proxy;
+    }
+    
+    /// <summary>
```
Comment (deeringc) [inline Release/include/cpprest/oauth2.h]: new accessors.

File: Release/src/http/oauth/oauth2.cpp [hunk 3]
```diff
@@ -101,6 +101,11 @@ pplx::task<void> oauth2_config::_request_token(uri_builder& request_body_ub)
     http_request request;
     request.set_method(methods::POST);
     request.set_request_uri(utility::string_t());
+    
+    if(!user_agent().empty())
```
Comment (deeringc) [inline Release/src/http/oauth/oauth2.cpp]: Otherwise, the default user agent will be used

**Comment by ras0219-msft** [comment]
Looks good to me, thanks for the PR and sorry for the many month delay!

**Pull request merged**